ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,200 @@
+                    {
+                        char t = sortedC[j];
+                        sortedC[j] = sortedC[j + 1];
+                        sortedC[j + 1] = t;
+                    }
+                }
+            }
+            string sorted = new string(sortedC);
+            return sorted;
+        }
+        public static char[] Duplicate(string str)
+        {
+            char[] array = new char[str.Length / 2 + 1];
+            char character;
+            int iter = 0;
+            
+            for (int i = 0; i < str.Length - 1; i++)
+            {
+                character = str[i];
+                for (int j = i + 1; j < str.Length; j++)
+                {
+                    if (character == str[j])
+                    {
+                        array[iter] = character;
+                        iter++;
+                        break;
+                    }
+                }
+            }
+            Array.Resize(ref array, iter);
+            return array;
+        }
+
+        // count how many times 'lorem' appears in the text (case insensitive)
+        // find second index of the 'pulvinar' word
+        // insert the same text in the middle after the 250th word
+        public static void LoremIpsumGenerator(string loremIpsum) 
+        {
+            int loremCount;
+            int spaceCount = 1;
+
+            int secondIndex = SecondIndexOf(loremIpsum, "pulvinar").Item2;
+            int length = loremIpsum.Length;
+            for (int i = 0; i < loremIpsum.Length; i++)
+            {
+                if (loremIpsum[i] == ' ')
+                    spaceCount++;
+                if (spaceCount == 250)
+                {
+                    loremIpsum = loremIpsum.Insert(i, loremIpsum);
+                    break;
+                }
+            }
+
+            loremCount = LoremCount(loremIpsum);
+            Console.WriteLine($"Overal info about string: {Environment.NewLine}" +
+                $" Lorem count: {loremCount} {Environment.NewLine}" +
+                $" Second index of 'pulvinar': {secondIndex} {Environment.NewLine}" +
+                $" New lorem ipsum before and inserted: {length} {loremIpsum.Length}");
+        }
+
+        public static (int, int) SecondIndexOf(string lorem, string word)
+        {
+            int firstIndex = lorem.IndexOf(word);
+            if (firstIndex == -1)
+                return (-1, -1);
+            string subFind = lorem.Substring(firstIndex + 1);
+            int secondIndex = subFind.IndexOf(word);
+            Console.WriteLine($"First index: {firstIndex}, second index: {secondIndex}");
+            return (firstIndex, secondIndex);
+        }
+
+        public static int LoremCount(string lorem)
+        {
+            int count = 0;
+            const string word = "lorem";
+            lorem = lorem.Replace(",", "");
+            lorem = lorem.Replace(".", "");
+            string[] words = lorem.Split(' ', StringSplitOptions.RemoveEmptyEntries);
```

Comment:
you can use split overload that accepts separators array, and you don't need to replace